lib/PAUSE/package.pm

```diff
@@ -257,21 +241,13 @@ sub examine_pkg {
 
   # should they be cought earlier? Maybe.
   # but as an ultimate sanity check suggested by Richard Soderberg
-  if ($self->_pkg_name_insane) {
-      $Logger->log("package[$package] name seems illegal");
-      delete $self->{FIO};    # circular reference
```

Comment:
We lost a lot of circular reference cleanup in this PR. Is this going to cause leaks?

Reply:
I don't _think_ so, but it almost doesn't matter:  mldistwatch does not make a lot of these, and it doesn't run persistently.  Still I'll give it a think.